applications/search/src/pages/informationmodel-details-page/model-component/model-component-property/model-component-property.component.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Scroll from 'react-scroll';
+
+import { getTypeFromTypeRef } from '../../../../lib/informationmodel';
+
+const ScrollLink = Scroll.Link;
+
+const createTypeRefScrollLink = typeRef => (
+  <ScrollLink
+    key={typeRef}
+    to={typeRef}
+    spy
+    smooth
+    isDynamic
+    offset={-10}
+    duration={1500}
+    tabIndex="0"
+  >
+    <span>{typeRef}</span>
+  </ScrollLink>
+);
+
+export const ModelComponentProperty = props => {
+  const { name, type, typeRef, oneOfTypeRefs } = props;
+
+  if (!(name || type || typeRef)) {
+    return null;
+  }
+
+  const typeRefArray = [];
+
+  if (oneOfTypeRefs) {
+    oneOfTypeRefs.forEach((ref, index) => {
+      typeRefArray.push(createTypeRefScrollLink(getTypeFromTypeRef(ref)));
+      if (index < oneOfTypeRefs.length) {
+        typeRefArray.push(' | ');
+      }
+    });
+  } else if (typeRef) {
+    typeRefArray.push(createTypeRefScrollLink(getTypeFromTypeRef(typeRef)));
+  }
+
+  return (
+    <div
+      style={{ marginLeft: '16px' }}
+      className="d-flex schema-collapse--item"
+      name={name}
+    >
+      <div className="w-50 pl-0">{name}</div>
+      <div className="w-50">
+        {(typeRefArray.length > 0 ? typeRefArray : null) || type}
```

Comment:
jeg anbefaler å samle array av data (uten render | inni data)
og her, i render blokk, har to altenariver. 
typeArray.length>1?( "("+typeArray.join('|')+")") : typeArray[0]

jeg vet ikke hvis det er lovlig i react :) fordi din typeArray er jo rendering.komponenter...
Får man ikke key error, hvis du render en array?


kanskje enkleste er til å separe de tre scenarioer til hver sit eget små funksjon. og ta vare et array av 1 item er samme som en...

også, det er kanskje godt mulig at oneOfTypeRefs er faktisk array av native typer (det kan være at min konversionfunksjon har ikke den variant dekket...